src/test/spec/unified_runner/operation.rs

```diff
@@ -1891,8 +1901,59 @@ impl TestOperation for AbortTransaction {
         test_runner: &'a TestRunner,
     ) -> BoxFuture<'a, Result<Option<Entity>>> {
         async move {
-            with_mut_session!(test_runner, id, |session| {
-                async move { session.abort_transaction().await }
+            with_mut_session!(test_runner, id, |session| async move {
+                session.abort_transaction().await
+            })
+            .await?;
+            Ok(None)
+        }
+        .boxed()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub(super) struct WithTransaction {
+    callback: Vec<Operation>,
+    #[serde(flatten)]
+    options: Option<TransactionOptions>,
+}
+
+impl TestOperation for WithTransaction {
+    fn execute_entity_operation<'a>(
+        &'a self,
+        id: &'a str,
+        test_runner: &'a TestRunner,
+    ) -> BoxFuture<'a, Result<Option<Entity>>> {
+        async move {
+            with_mut_session!(test_runner, id, |session| async move {
+                session
+                    .start_transaction()
+                    .with_options(self.options.clone())
+                    .and_run(
+                        (&self.callback, test_runner),
+                        |session, (callback, test_runner)| {
+                            async move {
+                                test_runner.entities.write().await.insert(
```

Comment:
This is the crux of the entity issue:
* SessionEntity is an owned value, but the callback to `start_transaction().and_run` (necessarily) receives a `&mut ClientSession`
* For the child operations to be able to access the session, it needs to be in the session map
* There's no expressable lifetime for the entity map to be parameterized on

To work around all of that, `WithTransaction` provides a raw pointer session for the entity map that `with_mut_session` and `as_mut_session` translate back into a `&mut`.